x-pack/docs/en/security/authentication/oidc-guide.asciidoc

```diff
@@ -0,0 +1,643 @@
+an {es} access token to be used as credentials for subsequent calls to {es} and a
+refresh token allowing to get new {es} access tokens for the user once the current one
+expires.
+
+NOTE: The {es} Token Service can be seen as a minimal oAuth2 authorization server
+and the access token and refresh token mentioned above are tokens that pertain
+_only_ to this authorization server, are generated and consumed _only_ by {es}
+and are in no way related to the tokens ( access token and ID Token ) that the
+OpenID Connect Provider issues.
+
+==== Register the RP with an OpenID Connect Provider
+
+The Relying Party ( {es} and the custom web app ) will need to be registered as
+client with the OpenID Connect Provider. Note that when registering the
+`Redirect URI`, it needs to be a URL in the custom web app.
+
+==== OpenID Connect Realm
+
+As before, an OpenID Connect realm needs to be created and configured accordingly
+in {es}. See <<oidc-guide-authentication>>
+
+==== Service Account user for accessing the APIs
+
+The realm is designed with the assumption that there needs to be a privileged entity
+acting as an authentication proxy. In this case, the custom web application is the
+authentication proxy handling the authentication of end users ( more correctly,
+"delegating" the authentication to the OpenID Connect Provider ). The OpenID Connect
+APIs require authentication and the necessary authorization level for the authenticated
+user. For this reason, a Service Account user needs to be created and assigned a role
+that gives them the `manage_oidc` cluster privilege. The use of the `manage_token`
+cluster privilege will be necessary after the authentication takes place, so that the
+the user can maintain access or be subsequently logged out.
+
+[source,js]
+--------------------------------------------------
+POST /_security/role/facilitator-role
+{
+  "cluster" : ["manage_oidc", "manage_token"]
+}
+--------------------------------------------------
+// CONSOLE
+
+
+[source,js]
+--------------------------------------------------
+POST /_security/user/facilitator
+{
+  "password" : "<somePasswordHere>",
+  "roles"    : [ "facilitator-role"]
+}
+--------------------------------------------------
+// CONSOLE
+
+
+==== Handling the authentication flow
+
+On a high level, the custom web application would need to perform the following steps in order to
+authenticate a user with OpenID Connect:
+
+. Make an HTTP POST request to `_security/oidc/prepare`, authenticating as the `facilitator` user, using the name of the
+OpenID Connect realm in the {es} configuration in the request body. See the
+<<security-api-oidc-prepare-authentication,OIDC Prepare Authentication API>> for more details
++
+[source,js]
+--------------------------------------------------
+POST /_security/oidc/prepare
+{
+  "realm" : "oidc1"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[skip:These are properly tested in the OpenIDConnectIT suite]
```

Comment:
Ideally we wouldn't skip these.
The intent of the docs-snippet testing is not to test the backend functionatlity, but to ensure that the docs content is free of errors, and is kept in sync with the implementation.
When we skip these, we run the risk of the examples being broken (e.g. by changes to the endpoint URL, or the addition of required fields)

Reply:
Understood. I'll take care of it

Reply:
I enabled tests for prepare and added `// TESTRESPONSE[catch:X]` for `authetnicate` and `logout` as there is no easy way to test them here. 

In order to get a succesful response from the integTest cluster here, we'd need to use the implicit flow and introduce code to sign the ID Token in a `// TESTSETUP` section so that the cluster can consume it. I don't think it's worth the effort, but I can be swayed